hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java

```diff
@@ -742,6 +743,12 @@ public static boolean isHqlTuple(SqmSelection<?> selection) {
 		return selection != null && selection.getSelectableNode() instanceof SqmTuple;
 	}
 
+	public static Type resolveExpressibleJavaTypeClass(final SqmExpression<?> expression) {
+		return expression.getExpressible() == null
+				? null
+				: expression.getExpressible().getExpressibleJavaType().getJavaType();
+	}
```

Comment:
Note that `expressible.getExpressibleJavaType()` can also return `null`. Also, please change the return type to reflect the method's name:

```suggestion
	public static Class<?> resolveExpressibleJavaTypeClass(final SqmExpression<?> expression) {
		final SqmExpressible<?> expressible = expression.getExpressible();
		return expressible != null && expressible.getExpressibleJavaType() != null ?
				expressible.getExpressibleJavaType().getJavaTypeClass() :
				null;
	}
```

Reply:
You're still returning `Type`, please change it to `Class<?>`. Also please don't call `getExpressible()` multiple times as in some cases there's some processing involved, if you could apply my suggestion it would be great.